dali/python/nvidia/dali/triton.py

```diff
@@ -0,0 +1,70 @@
+# Copyright (c) 2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import sys
+
+pipeline_autoserialized = False
+
+
+def autoserialize(dali_pipeline):
+    """
+    Decorator, that marks a DALI pipeline (represented by :meth:`nvidia.dali.pipeline_def`) for
+    autoserialization in DALI Backend's model repository.
+
+    For details about the autoserialization feature, please refer to the DALI Backend documentation.
+
+    To properly autoserialize the DALI pipeline, the caller needs to run the Python script, that
+    contains the pipeline marked with ``autoserialize``, pass the target file name
+    (i.e. the path to the file, where the serialized model will be saved) and the magical command
+    ``autoserialize.me``. For example:
+
+        # identity.py
+        import nvidia.dali as dali
+
+        @dali.triton.autoserialize
+        @dali.pipeline_def(batch_size=3, num_threads=1, device_id=0)
+        def pipe():
+            data = dali.fn.external_source(device="cpu", name="DALI_INPUT_0")
+            return data
+
+
+        $ python identity.py /tmp/model.dali autoserialize.me
```

Comment:
You sure about that usage?

As an alternative I would suggest trying to import that as a module (so a safeguard against someone putting actual `__main__` section), and looking for a function that would be marked for export by the autoserialize.

Another alternative would be to look how tools like nose does the test case discovery and doing something similar.

Reply:
Do I understand correctly that you propose to do something along:
```python
# mydali.py
@autoserialize
@pipeline_def
def mypipe():
   ...
```
```bash
$ python -c 'import mydali; f = discover_autoserialize(); f()'
```
This actually might work. I'll try to do it this way. And you're right we should secure it from actually putting the `__main__` magic.

About how does nose discover tests - it uses the name (i.e. when a function contains a `test` word) and the special attribute: `__test__`.

Reply:
Something like this. You can probably look for whatever global symbol has the some magic member set by autoserialize or some similar mechanism.

Reply:
Done! How about now?